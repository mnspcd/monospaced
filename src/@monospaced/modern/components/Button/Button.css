.mdrn .Button {
  background-color: var(--color-blue-6);
  border-color: transparent;
  border-radius: 0.25em;
  border-style: solid;
  border-width: calc(1rem / 16);
  color: var(--color-white);
  font-family: var(--font-functional);
  font-size: 1em;
  -webkit-font-smoothing: auto;
  height: calc(var(--leading) * 1.5em);
  line-height: 1;
  min-width: calc(48rem / 16);
  -moz-osx-font-smoothing: auto;
  padding-left: 1em;
  padding-right: 1em;
  text-align: center;
  white-space: nowrap;
}

.mdrn .Button:hover {
  background-color: var(--color-blue-7);
  outline: 0;
}

.mdrn .Button:focus:not(:focus-visible) {
  outline: 0;
}

.mdrn .Button:active {
  background-color: var(--color-blue-8);
  outline: 0;
}

/* dark background */

.mdrn .Button--onDarkBackground {
  background-color: var(--color-blue-5);
  color: var(--color-grey-9);
}

.mdrn .Button--onDarkBackground:hover {
  background-color: var(--color-blue-4);
}

.mdrn .Button--onDarkBackground:active {
  background-color: var(--color-blue-3);
}

/* danger */

.mdrn .Button--colorDanger {
  background-color: var(--color-red-6);
}

.mdrn .Button--colorDanger:hover {
  background-color: var(--color-red-7);
}

.mdrn .Button--colorDanger:active {
  background-color: var(--color-red-8);
}

/* danger on dark background */

.mdrn .Button--colorDanger.Button--onDarkBackground {
  background-color: var(--color-red-5);
}

.mdrn .Button--colorDanger.Button--onDarkBackground:hover {
  background-color: var(--color-red-4);
}

.mdrn .Button--colorDanger.Button--onDarkBackground:active {
  background-color: var(--color-red-3);
}

/* flat
   ========================================================================== */

.mdrn .Button--styleFlat {
  background-color: transparent;
  color: var(--color-blue-6);
}

.mdrn .Button--styleFlat:hover {
  background-color: rgba(var(--color-blue-7-rgb), var(--opacity-hover));
  color: var(--color-blue-7);
}

.mdrn .Button--styleFlat:active {
  background-color: rgba(var(--color-blue-8-rgb), var(--opacity-active));
  color: var(--color-blue-8);
}

/* flat on dark background */

.mdrn .Button--styleFlat.Button--onDarkBackground {
  color: var(--color-blue-5);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.mdrn .Button--styleFlat.Button--onDarkBackground:hover {
  background-color: rgba(var(--color-blue-4-rgb), var(--opacity-hover));
  color: var(--color-blue-4);
}

.mdrn .Button--styleFlat.Button--onDarkBackground:active {
  background-color: rgba(var(--color-blue-3-rgb), var(--opacity-active));
  color: var(--color-blue-3);
}

/* flat danger */

.mdrn .Button--styleFlat.Button--colorDanger {
  background-color: transparent;
  color: var(--color-red-6);
}

.mdrn .Button--styleFlat.Button--colorDanger:hover {
  background-color: rgba(var(--color-red-7-rgb), var(--opacity-hover));
  color: var(--color-red-7);
}

.mdrn .Button--styleFlat.Button--colorDanger:active {
  background-color: rgba(var(--color-red-8-rgb), var(--opacity-active));
  color: var(--color-red-8);
}

/* flat danger on dark background */

.mdrn .Button--styleFlat.Button--colorDanger.Button--onDarkBackground {
  color: var(--color-red-5);
}

.mdrn .Button--styleFlat.Button--colorDanger.Button--onDarkBackground:hover {
  background-color: rgba(var(--color-red-4-rgb), var(--opacity-hover));
  color: var(--color-red-4);
}

.mdrn .Button--styleFlat.Button--colorDanger.Button--onDarkBackground:active {
  background-color: rgba(var(--color-red-3-rgb), var(--opacity-active));
  color: var(--color-red-3);
}

/* outlined
   ========================================================================== */

.mdrn .Button--styleOutlined {
  background-color: transparent;
  border-color: var(--color-grey-4);
  color: var(--color-blue-6);
}

.mdrn .Button--styleOutlined:hover {
  background-color: rgba(var(--color-blue-7-rgb), var(--opacity-hover));
  color: var(--color-blue-7);
}

.mdrn .Button--styleOutlined:active {
  background-color: rgba(var(--color-blue-8-rgb), var(--opacity-active));
  color: var(--color-blue-8);
}

/* outlined on dark background */

.mdrn .Button--styleOutlined.Button--onDarkBackground {
  border-color: var(--color-grey-7);
  color: var(--color-blue-5);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.mdrn .Button--styleOutlined.Button--onDarkBackground:hover {
  background-color: rgba(var(--color-blue-4-rgb), var(--opacity-hover));
  color: var(--color-blue-4);
}

.mdrn .Button--styleOutlined.Button--onDarkBackground:active {
  background-color: rgba(var(--color-blue-3-rgb), var(--opacity-active));
  color: var(--color-blue-3);
}

/* outlined danger */

.mdrn .Button--styleOutlined.Button--colorDanger {
  background-color: transparent;
  color: var(--color-red-6);
}

.mdrn .Button--styleOutlined.Button--colorDanger:hover {
  background-color: rgba(var(--color-red-7-rgb), var(--opacity-hover));
  color: var(--color-red-7);
}

.mdrn .Button--styleOutlined.Button--colorDanger:active {
  background-color: rgba(var(--color-red-8-rgb), var(--opacity-active));
  color: var(--color-red-8);
}

/* outlined danger on dark background */

.mdrn .Button--styleOutlined.Button--colorDanger.Button--onDarkBackground {
  color: var(--color-red-5);
}

.mdrn .Button--styleOutlined.Button--colorDanger.Button--onDarkBackground:hover {
  background-color: rgba(var(--color-red-4-rgb), var(--opacity-hover));
  color: var(--color-red-4);
}

.mdrn .Button--styleOutlined.Button--colorDanger.Button--onDarkBackground:active {
  background-color: rgba(var(--color-red-3-rgb), var(--opacity-active));
  color: var(--color-red-3);
}

/* on Poster
   ========================================================================== */

.mdrn .Button--onPoster {
  background-color: transparent;
  border: none;
  color: var(--color-white);
}

.mdrn .Button--onPoster:hover {
  background-color: rgba(var(--color-blue-7-rgb), var(--opacity-hover));
  color: var(--color-blue-2);
}

.mdrn .Button--onPoster:active {
  background-color: rgba(var(--color-blue-8-rgb), var(--opacity-active));
  color: var(--color-blue-3);
}

.mdrn .Button--onPoster.Button--colorDanger {
  color: var(--color-red-5);
}

.mdrn .Button--onPoster.Button--colorDanger:hover {
  background-color: rgba(var(--color-red-7-rgb), var(--opacity-hover));
  color: var(--color-red-4);
}

.mdrn .Button--onPoster.Button--colorDanger:active {
  background-color: rgba(var(--color-red-8-rgb), var(--opacity-active));
  color: var(--color-red-3);
}

/* disabled
   ========================================================================== */

.mdrn .Button:disabled,
.mdrn .Button--colorDanger:disabled {
  background-color: var(--color-grey-3);
  color: var(--color-grey-5);
}

.mdrn .Button.Button--onDarkBackground:disabled,
.mdrn .Button--colorDanger.Button--onDarkBackground:disabled {
  background-color: var(--color-grey-8);
  color: var(--color-grey-6);
}

.mdrn .Button--styleFlat:disabled,
.mdrn .Button--styleFlat.Button--colorDanger:disabled {
  background-color: transparent;
  color: var(--color-grey-4);
}

.mdrn .Button--styleFlat.Button--onDarkBackground:disabled,
.mdrn .Button--styleFlat.Button--colorDanger.Button--onDarkBackground:disabled {
  background-color: transparent;
  color: var(--color-grey-7);
}

.mdrn .Button--styleOutlined:disabled,
.mdrn .Button--styleOutlined.Button--colorDanger:disabled {
  background-color: transparent;
  border-color: var(--color-grey-3);
  color: var(--color-grey-4);
}

.mdrn .Button--styleOutlined.Button--onDarkBackground:disabled,
.mdrn .Button--styleOutlined.Button--colorDanger.Button--onDarkBackground:disabled {
  background-color: transparent;
  border-color: var(--color-grey-8);
  color: var(--color-grey-7);
}

.mdrn .Button--onPoster:disabled {
  background-color: transparent;
  color: var(--color-grey-4);
}

/* text
   ========================================================================== */

.mdrn .Button-text {
  position: relative;
}

.mdrn .Button--withIcon .Button-text {
  margin-left: 0.5em;
  margin-right: 0.5em;
  vertical-align: middle;
}

/* icon
   ========================================================================== */

.mdrn .Button-icon {
  display: inline-block;
  height: calc(20rem / 16);
  position: relative;
  vertical-align: middle;
  width: calc(20rem / 16);
}

@media (--larger-viewport) {
  .mdrn .Button-icon {
    height: calc(24rem / 16);
    width: calc(24rem / 16);
  }
}

.mdrn .Button--iconOnly {
  height: calc(48rem / 16);
  min-width: auto;
  padding-left: 0;
  padding-right: 0;
  width: calc(48rem / 16);
}

.mdrn .Button--iconOnly .Button-icon {
  height: calc(24rem / 16);
  width: calc(24rem / 16);
}

@media (--larger-viewport) {
  .mdrn .Button--iconOnly {
    height: calc(45rem / 16);
    width: calc(45rem / 16);
  }
}
