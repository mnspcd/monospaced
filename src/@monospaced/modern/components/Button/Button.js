import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { PosterConsumer } from "../Poster";
import { SurfaceConsumer } from "../Surface";

import "./Button.css";

/**
 * Use `Button` to allow users to take actions with clicks and taps.
 */
const Button = ({
  children = null,
  colorVariant = null,
  disabled = null,
  icon = null,
  minWidth = null,
  styleVariant = null,
}) => {
  return (
    <PosterConsumer>
      {poster => (
        <SurfaceConsumer>
          {surfaceBackground => (
            <button
              className={classNames({
                Button: true,
                [`Button--colorDanger`]: colorVariant === "danger",
                [`Button--styleFlat`]: styleVariant === "flat",
                [`Button--styleOutlined`]: styleVariant === "outlined",
                [`Button--withIcon`]: icon,
                [`Button--iconOnly`]: icon && !children,
                [`Button--onDarkBackground`]: surfaceBackground === "dark",
                [`Button--onPoster`]: poster,
              })}
              disabled={disabled}
              style={{ minWidth }}
            >
              {icon && <span className="Button-icon">{icon}</span>}
              {children && <span className="Button-text">{children}</span>}
            </button>
          )}
        </SurfaceConsumer>
      )}
    </PosterConsumer>
  );
};

const colorVariants = ["danger"];
const styleVariants = ["flat", "outlined"];

Button.colorVariants = colorVariants;
Button.styleVariants = styleVariants;

Button.propTypes = {
  /**
   * Text to display in the Button.
   */
  children: PropTypes.string,

  /**
   * Semantic Button color variants.
   */
  colorVariant: PropTypes.oneOf(colorVariants),

  /**
   * Is the Button disabled?
   */
  disabled: PropTypes.bool,

  /**
   * Icon to display in the Button. Usually a react-feather Icon component with
   * size prop value of 100%.
   */
  icon: PropTypes.element,

  /**
   * A valid CSS min-width value for the Button.
   */
  minWidth: PropTypes.string,

  /**
   * Visual Button style variants.
   */
  styleVariant: PropTypes.oneOf(styleVariants),
};

export default Button;
