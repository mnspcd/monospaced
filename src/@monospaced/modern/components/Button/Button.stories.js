import { boolean, select, text } from "@storybook/addon-knobs";
import React from "react";
import * as Icons from "react-feather";

import Group from "../Group";
import Poster from "../Poster";
import Space from "../Space";
import Surface from "../Surface";

import Button from "./Button";

export default {
  component: Button,
  title: "Components/Button",
};

export const basic = () => (
  <Group>
    <Group.Item>
      <Button icon={<Icons.Star size="100%" />}>Default</Button>
    </Group.Item>
    <Group.Item>
      <Button icon={<Icons.Star size="100%" />} styleVariant="outlined">
        Outlined
      </Button>
    </Group.Item>
    <Group.Item>
      <Button icon={<Icons.Star size="100%" />} styleVariant="flat">
        Flat
      </Button>
    </Group.Item>
  </Group>
);

export const textOnly = () => (
  <Group>
    <Group.Item>
      <Button>Default</Button>
    </Group.Item>
    <Group.Item>
      <Button styleVariant="outlined">Outlined</Button>
    </Group.Item>
    <Group.Item>
      <Button styleVariant="flat">Flat</Button>
    </Group.Item>
  </Group>
);

export const iconOnly = () => (
  <Group>
    <Group.Item>
      <Button icon={<Icons.Star size="100%" />} />
    </Group.Item>
    <Group.Item>
      <Button icon={<Icons.Star size="100%" />} styleVariant="outlined" />
    </Group.Item>
    <Group.Item>
      <Button icon={<Icons.Star size="100%" />} styleVariant="flat" />
    </Group.Item>
  </Group>
);

export const danger = () => (
  <Group>
    <Group.Item>
      <Button colorVariant="danger" icon={<Icons.Star size="100%" />}>
        Default
      </Button>
    </Group.Item>
    <Group.Item>
      <Button
        colorVariant="danger"
        icon={<Icons.Star size="100%" />}
        styleVariant="outlined"
      >
        Outlined
      </Button>
    </Group.Item>
    <Group.Item>
      <Button
        colorVariant="danger"
        icon={<Icons.Star size="100%" />}
        styleVariant="flat"
      >
        Flat
      </Button>
    </Group.Item>
  </Group>
);

export const disabled = () => (
  <Group>
    <Group.Item>
      <Button disabled icon={<Icons.Star size="100%" />}>
        Default
      </Button>
    </Group.Item>
    <Group.Item>
      <Button
        disabled
        icon={<Icons.Star size="100%" />}
        styleVariant="outlined"
      >
        Outlined
      </Button>
    </Group.Item>
    <Group.Item>
      <Button disabled icon={<Icons.Star size="100%" />} styleVariant="flat">
        Flat
      </Button>
    </Group.Item>
  </Group>
);

export const onDarkBackground = () => (
  <Surface backgroundColor="grey-dark">
    <div style={{ padding: "1em" }}>
      <Group>
        <Group.Item>
          <Button icon={<Icons.Star size="100%" />}>Default</Button>
        </Group.Item>
        <Group.Item>
          <Button icon={<Icons.Star size="100%" />} styleVariant="outlined">
            Outlined
          </Button>
        </Group.Item>
        <Group.Item>
          <Button icon={<Icons.Star size="100%" />} styleVariant="flat">
            Flat
          </Button>
        </Group.Item>
      </Group>
      <Space />
      <Group>
        <Group.Item>
          <Button colorVariant="danger" icon={<Icons.Star size="100%" />}>
            Default
          </Button>
        </Group.Item>
        <Group.Item>
          <Button
            colorVariant="danger"
            icon={<Icons.Star size="100%" />}
            styleVariant="outlined"
          >
            Outlined
          </Button>
        </Group.Item>
        <Group.Item>
          <Button
            colorVariant="danger"
            icon={<Icons.Star size="100%" />}
            styleVariant="flat"
          >
            Flat
          </Button>
        </Group.Item>
      </Group>
      <Space />
      <Group>
        <Group.Item>
          <Button disabled icon={<Icons.Star size="100%" />}>
            Default
          </Button>
        </Group.Item>
        <Group.Item>
          <Button
            disabled
            icon={<Icons.Star size="100%" />}
            styleVariant="outlined"
          >
            Outlined
          </Button>
        </Group.Item>
        <Group.Item>
          <Button
            disabled
            icon={<Icons.Star size="100%" />}
            styleVariant="flat"
          >
            Flat
          </Button>
        </Group.Item>
      </Group>
    </div>
  </Surface>
);

export const onPoster = () => (
  <Poster padding="1em">
    <Button icon={<Icons.Star size="100%" />}>Default</Button>
    <Space size="s" />
    <Button colorVariant="danger" icon={<Icons.Star size="100%" />}>
      Default
    </Button>
    <Space size="s" />
    <Button disabled icon={<Icons.Star size="100%" />}>
      Default
    </Button>
  </Poster>
);

export const fullWidth = () => (
  <>
    <Button icon={<Icons.Star size="100%" />} minWidth="100%">
      Default
    </Button>
    <Space size="s" />
    <Button
      icon={<Icons.Star size="100%" />}
      minWidth="100%"
      styleVariant="outlined"
    >
      Outlined
    </Button>
    <Space size="s" />
    <Button
      icon={<Icons.Star size="100%" />}
      minWidth="100%"
      styleVariant="flat"
    >
      Flat
    </Button>
  </>
);

export const knobs = () => {
  const buttonText = text("text", "Button", "Button") || null;
  const Icon =
    Icons[
      select("icon", [""].concat(Object.keys(Icons).sort()), "Star", "Button")
    ];
  const colorVariant =
    select("colorVariant", [""].concat(Button.colorVariants), "", "Button") ||
    null;
  const styleVariant =
    select("styleVariant", [""].concat(Button.styleVariants), "", "Button") ||
    null;
  const minWidth = text("minWidth", "", "Button") || null;
  const isDisabled = boolean("disabled", false, "Button");

  return (
    <Surface
      backgroundColor={select(
        "backgroundColor",
        Surface.backgroundColors,
        "white",
        "Surface",
      )}
    >
      <Button
        colorVariant={colorVariant}
        styleVariant={styleVariant}
        minWidth={minWidth}
        disabled={isDisabled}
        icon={Icon && <Icon size="100%" />}
      >
        {buttonText}
      </Button>
    </Surface>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
