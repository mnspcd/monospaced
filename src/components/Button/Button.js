import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import { PosterConsumer } from "../Poster";
import { SurfaceConsumer } from "../Surface";
import "./Button.css";

const Button = ({
  children,
  colorVariant,
  disabled,
  icon,
  minWidth,
  styleVariant,
}) => {
  return (
    <PosterConsumer>
      {poster => (
        <SurfaceConsumer>
          {surfaceBackground => (
            <button
              className={classNames({
                Button: true,
                [`Button--colorDanger`]: colorVariant === "danger",
                [`Button--styleFlat`]: styleVariant === "flat",
                [`Button--styleOutlined`]: styleVariant === "outlined",
                [`Button--withIcon`]: icon,
                [`Button--iconOnly`]: icon && !children,
                [`Button--onDarkBackground`]: surfaceBackground === "dark",
                [`Button--onPoster`]: poster,
              })}
              disabled={disabled}
              style={{ minWidth }}
            >
              {icon && <span className="Button-icon">{icon}</span>}
              {children && <span className="Button-text">{children}</span>}
            </button>
          )}
        </SurfaceConsumer>
      )}
    </PosterConsumer>
  );
};

Button.colorVariants = ["danger"];
Button.styleVariants = ["flat", "outlined"];

Button.propTypes = {
  children: PropTypes.node,
  colorVariant: PropTypes.oneOf(Button.colorVariants),
  disabled: PropTypes.bool,
  icon: PropTypes.node,
  minWidth: PropTypes.string,
  styleVariant: PropTypes.oneOf(Button.styleVariants),
};

export default Button;
