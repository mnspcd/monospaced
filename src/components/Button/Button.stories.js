import React from "react";
import { boolean, select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as Icons from "react-feather";

import Grid from "../Grid";
import Group from "../Group";
import Space from "../Space";
import Surface from "../Surface";

import Button from "./Button";

storiesOf("Components/Button", module)
  .add("Component", () => {
    const buttonText = text("text", "Make it rain", "Button") || null;
    const Icon =
      Icons[
        select(
          "icon",
          [""].concat(Object.keys(Icons).sort()),
          "CloudRain",
          "Button",
        )
      ];
    const colorVariant =
      select("colorVariant", [""].concat(Button.colorVariants), "", "Button") ||
      null;
    const styleVariant =
      select("styleVariant", [""].concat(Button.styleVariants), "", "Button") ||
      null;
    const minWidth = text("minWidth", "", "Button") || null;
    const disabled = boolean("disabled", false, "Button");

    return (
      <Surface
        backgroundColor={select(
          "backgroundColor",
          Surface.backgroundColors,
          "white",
          "Surface",
        )}
      >
        <Button
          colorVariant={colorVariant}
          styleVariant={styleVariant}
          minWidth={minWidth}
          disabled={disabled}
          icon={Icon && <Icon size="100%" />}
        >
          {buttonText}
        </Button>
      </Surface>
    );
  })
  .add("Variants", () => {
    const buttonText = text("text", "Make it rain", "Button") || null;
    const Icon =
      Icons[
        select(
          "icon",
          [""].concat(Object.keys(Icons).sort()),
          "CloudRain",
          "Button",
        )
      ];

    return (
      <Surface
        backgroundColor={select(
          "backgroundColor",
          Surface.backgroundColors,
          "white",
          "Surface",
        )}
      >
        <Grid>
          <Grid.Item colSpan="12">
            <Space />
            {buttonText && Icon && (
              <Space>
                <Space>
                  <Group>
                    <Group.Item>
                      <Button icon={Icon && <Icon size="100%" />}>
                        {buttonText}
                      </Button>
                    </Group.Item>
                    <Group.Item>
                      <Button
                        icon={Icon && <Icon size="100%" />}
                        styleVariant="outlined"
                      >
                        {buttonText}
                      </Button>
                    </Group.Item>
                    <Group.Item>
                      <Button
                        icon={Icon && <Icon size="100%" />}
                        styleVariant="flat"
                      >
                        {buttonText}
                      </Button>
                    </Group.Item>
                  </Group>
                </Space>
                <Space>
                  <Group>
                    <Group.Item>
                      <Button
                        colorVariant="danger"
                        icon={Icon && <Icon size="100%" />}
                      >
                        {buttonText}
                      </Button>
                    </Group.Item>
                    <Group.Item>
                      <Button
                        colorVariant="danger"
                        icon={Icon && <Icon size="100%" />}
                        styleVariant="outlined"
                      >
                        {buttonText}
                      </Button>
                    </Group.Item>
                    <Group.Item>
                      <Button
                        colorVariant="danger"
                        icon={Icon && <Icon size="100%" />}
                        styleVariant="flat"
                      >
                        {buttonText}
                      </Button>
                    </Group.Item>
                  </Group>
                </Space>
                <Group>
                  <Group.Item>
                    <Button icon={Icon && <Icon size="100%" />} disabled>
                      {buttonText}
                    </Button>
                  </Group.Item>
                  <Group.Item>
                    <Button
                      icon={Icon && <Icon size="100%" />}
                      styleVariant="outlined"
                      disabled
                    >
                      {buttonText}
                    </Button>
                  </Group.Item>
                  <Group.Item>
                    <Button
                      icon={Icon && <Icon size="100%" />}
                      styleVariant="flat"
                      disabled
                    >
                      {buttonText}
                    </Button>
                  </Group.Item>
                </Group>
              </Space>
            )}
            {buttonText && (
              <Space>
                <Space>
                  <Group>
                    <Group.Item>
                      <Button>{buttonText}</Button>
                    </Group.Item>
                    <Group.Item>
                      <Button styleVariant="outlined">{buttonText}</Button>
                    </Group.Item>
                    <Group.Item>
                      <Button styleVariant="flat">{buttonText}</Button>
                    </Group.Item>
                  </Group>
                </Space>
                <Space>
                  <Group>
                    <Group.Item>
                      <Button colorVariant="danger">{buttonText}</Button>
                    </Group.Item>
                    <Group.Item>
                      <Button colorVariant="danger" styleVariant="outlined">
                        {buttonText}
                      </Button>
                    </Group.Item>
                    <Group.Item>
                      <Button colorVariant="danger" styleVariant="flat">
                        {buttonText}
                      </Button>
                    </Group.Item>
                  </Group>
                </Space>
                <Group>
                  <Group.Item>
                    <Button disabled>{buttonText}</Button>
                  </Group.Item>
                  <Group.Item>
                    <Button styleVariant="outlined" disabled>
                      {buttonText}
                    </Button>
                  </Group.Item>
                  <Group.Item>
                    <Button styleVariant="flat" disabled>
                      {buttonText}
                    </Button>
                  </Group.Item>
                </Group>
              </Space>
            )}
            {Icon && (
              <Space>
                <Space>
                  <Group>
                    <Group.Item>
                      <Button icon={<Icon size="100%" />} />
                    </Group.Item>
                    <Group.Item>
                      <Button
                        icon={<Icon size="100%" />}
                        styleVariant="outlined"
                      />
                    </Group.Item>
                    <Group.Item>
                      <Button icon={<Icon size="100%" />} styleVariant="flat" />
                    </Group.Item>
                  </Group>
                </Space>
                <Space>
                  <Group>
                    <Group.Item>
                      <Button
                        colorVariant="danger"
                        icon={<Icon size="100%" />}
                      />
                    </Group.Item>
                    <Group.Item>
                      <Button
                        colorVariant="danger"
                        icon={<Icon size="100%" />}
                        styleVariant="outlined"
                      />
                    </Group.Item>
                    <Group.Item>
                      <Button
                        colorVariant="danger"
                        icon={<Icon size="100%" />}
                        styleVariant="flat"
                      />
                    </Group.Item>
                  </Group>
                </Space>
                <Group>
                  <Group.Item>
                    <Button icon={<Icon size="100%" />} disabled />
                  </Group.Item>
                  <Group.Item>
                    <Button
                      icon={<Icon size="100%" />}
                      styleVariant="outlined"
                      disabled
                    />
                  </Group.Item>
                  <Group.Item>
                    <Button
                      icon={<Icon size="100%" />}
                      styleVariant="flat"
                      disabled
                    />
                  </Group.Item>
                </Group>
              </Space>
            )}
          </Grid.Item>
        </Grid>
      </Surface>
    );
  });
