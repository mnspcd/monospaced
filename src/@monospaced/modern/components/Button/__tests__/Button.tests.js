/* eslint-env jest */
import { mount, render } from "enzyme";
import React from "react";

import Button from "../Button";

describe("Button component", () => {
  it("should render correctly", () => {
    const component = render(<Button />);
    expect(component).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    const component = render(<Button>children</Button>);
    expect(component).toMatchSnapshot();
  });

  it("should render an icon correctly", () => {
    const component = render(<Button icon={<div />}>children</Button>);
    expect(component).toMatchSnapshot();
  });

  it("should render an icon and no children correctly", () => {
    const component = render(<Button icon={<div />} />);
    expect(component).toMatchSnapshot();
  });

  it("should handle default onClick function correctly", () => {
    const component = mount(<Button icon={<div />} />);

    component.simulate("click");

    expect(component).toMatchSnapshot();
  });

  it("should call onClick function correctly", () => {
    const mockFunction = jest.fn();

    const component = mount(<Button icon={<div />} onClick={mockFunction} />);

    expect(mockFunction).not.toHaveBeenCalled();

    component.simulate("click");

    expect(mockFunction).toHaveBeenCalledTimes(1);
  });
});
