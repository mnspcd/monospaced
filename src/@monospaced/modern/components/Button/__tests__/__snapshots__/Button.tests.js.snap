// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should handle default onClick function correctly 1`] = `
<Button
  icon={<div />}
>
  <button
    className="Button Button--withIcon Button--iconOnly"
    disabled={null}
    onClick={[Function]}
    style={
      Object {
        "minWidth": null,
      }
    }
  >
    <span
      className="Button-icon"
    >
      <div />
    </span>
  </button>
</Button>
`;

exports[`Button component should render an icon and no children correctly 1`] = `
<button
  class="Button Button--withIcon Button--iconOnly"
>
  <span
    class="Button-icon"
  >
    <div />
  </span>
</button>
`;

exports[`Button component should render an icon correctly 1`] = `
<button
  class="Button Button--withIcon"
>
  <span
    class="Button-icon"
  >
    <div />
  </span>
  <span
    class="Button-text"
  >
    children
  </span>
</button>
`;

exports[`Button component should render children correctly 1`] = `
<button
  class="Button"
>
  <span
    class="Button-text"
  >
    children
  </span>
</button>
`;

exports[`Button component should render correctly 1`] = `
<button
  class="Button"
/>
`;
